integration/containerd/runk/runk-tests.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+#
+# Copyright (c) 2021-2022 Kata Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will validate runk with containerd
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+dir_path=$(dirname "$0")
+source /etc/os-release || source /usr/lib/os-release
+source "${dir_path}/../../../lib/common.bash"
+source "${dir_path}/../../../.ci/lib.sh"
+
+KATACONTAINERS_REPO=${katacontainers_repo:="github.com/kata-containers/kata-containers"}
+RUNK_SRC_PATH="${GOPATH}/src/${KATACONTAINERS_REPO}/src/tools/runk"
+RUNK_BIN_PATH="/usr/local/bin/runk"
+TEST_IMAGE="docker.io/library/busybox:latest"
+CONTAINER_ID="id1"
+PID_FILE="${CONTAINER_ID}.pid"
+
+setup() {
+    # can't find cargo when make runk below, so we install rust to make runk build correctly
+    "${dir_path}/../../../.ci/install_rust.sh" && source "$HOME/.cargo/env"
+    echo "restart containerd service"
+    sudo systemctl restart containerd
+    echo "pull container image"
+    sudo ctr images pull ${TEST_IMAGE}
+}
+
+install_runk() {
+    echo "Install runk"
+    pushd ${RUNK_SRC_PATH}
+    make
+    sudo make install
+    popd
+}
+
+test_runk() {
+    echo "start container with runk"
+    sudo ctr run --pid-file ${PID_FILE} --rm -d --runc-binary ${RUNK_BIN_PATH} ${TEST_IMAGE} ${CONTAINER_ID}
+    read CID PID STATUS <<< $(sudo ctr t ls | grep ${CONTAINER_ID})
+    [ ${PID} == $(cat ${PID_FILE}) ] || die "pid is not consistent"
+    [ ${STATUS} == "RUNNING" ] || die "contianer status is not RUNNING"
+
+    echo "exec process in a container"
+    sudo ctr t exec --exec-id id1 ${CONTAINER_ID} sh -c "echo hello > /tmp/foo"
+    [ "hello" == "$(sudo ctr t exec --exec-id id1 ${CONTAINER_ID} cat /tmp/foo)" ] || die "exec process failed"
+
+    echo "kill and delete container"
+    sudo ctr t kill --signal SIGKILL --all ${CONTAINER_ID}
+    # sleep for a while to make sure the task is killed
+    sleep 5s
```

Comment:
`ctr` has a global option `--timeout`, that could prevent time wasting by `sleep 5s`

Reply:
`sudo ctr t kill --signal SIGKILL --all ${CONTAINER_ID}` to `sudo ctr --timeout 5s t kill --signal SIGKILL --all ${CONTAINER_ID}`?

Reply:
any reason of not using https://github.com/kata-containers/tests/blob/main/lib/common.bash#L221 @cyyzero ?

Reply:
@GabyCT Thanks for reminding me of this function! I will use it.

Reply:
Only SIGKILL or SIGSTOP can ensure stop the init process in a pid namespace.

According to [man 7 pid_namespaces](https://man7.org/linux/man-pages/man7/pid_namespaces.7.html), only SIGKILL or SIGSTOP can ensure stop the init process in a pid namespace.:

>        Likewise, a process in an ancestor namespace can—subject to the
>        usual permission checks described in [kill(2)](https://man7.org/linux/man-pages/man2/kill.2.html)—send signals to the
>        "init" process of a child PID namespace only if the "init"
>        process has established a handler for that signal.  (Within the
>        handler, the siginfo_t si_pid field described in [sigaction(2)](https://man7.org/linux/man-pages/man2/sigaction.2.html)
>        will be zero.)  SIGKILL or SIGSTOP are treated exceptionally:
>        these signals are forcibly delivered when sent from an ancestor
>        PID namespace.  Neither of these signals can be caught by the
>        "init" process, and so will result in the usual actions
>        associated with those signals (respectively, terminating and
>        stopping the process).

`Ctr kill` send SIGTERM by default. There is a high probability that the container process( also as the init process in PID namespace) does not establish a handler for SIGTERM. So if `ctr t kill` command (which is a process in an ancestor PID namespace) send SIGTERM, container process will not receive.

In conclusion, here should use `ctr t kill --signal SIGKILL --all`. Is it ok to update clean_env and add these options for `ctr t kill`?  I'm not sure if this will affect kata's tests.